migration/register.go

```diff
@@ -0,0 +1,120 @@
+package migration
+
+import (
+	"reflect"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+// Payload is implemented by both weave.Msg and models used by the orm package.
+// Schema migration supports both of those data types.
+type Payload interface {
+	GetHeader() *weave.Header
+	Validate() error
+}
+
+// Migrator is a function that migrates a data entity from version
+// requiredVersion-1 to requested version.
+type Migrator func(ctx weave.Context, db weave.KVStore, msgOrModel Payload) error
+
+// NoModyfication is a migration function that migrates data that requires no
+// change. It should be used to register migrations that do not require any
+// modyfications.
+func NoModyfication(ctx weave.Context, db weave.KVStore, msgOrModel Payload) error {
+	return nil
+}
+
+func newRegister() *register {
+	return &register{
+		handlers: make(map[payloadVersion]Migrator),
+	}
+}
+
+type register struct {
+	handlers map[payloadVersion]Migrator
+}
+
+// payloadVersion references a message or a model at a given schema version.
+type payloadVersion struct {
+	payload reflect.Type
```

Comment:
Really?? You can use this in the key for a map???

Reply:
It's not the `reflect.Type` that is the key but a `payloadVersion` instance. Sure, why not? :muscle: